python/cudf/cudf/core/series.py

```diff
@@ -1652,7 +1652,16 @@ def label_encoding(self, cats, dtype=None, na_sentinel=-1):
         if dtype is None:
             dtype = min_scalar_type(len(cats), 8)
 
-        cats = Series(cats).astype(self.dtype)
+        cats = Series(cats)
+        try:
+            # Where there is a type-cast from string to numeric types,
+            # there is a possibility for ValueError when strings
+            # are having non-numeric values, hence we will have
+            # to catch the exception and fill in with zeros.
+            cats = cats.astype(self.dtype)
+        except ValueError:
+            cats = Series(cupy.zeros(len(cats), dtype=self.dtype))
```

Comment:
What does a categories of all zeros even mean? If we can't safely cast what's passed into `cats` to the dtype of what we're encoding we should probably just throw an informative error, no?

Reply:
Cats with all zeros would result in all nulls for the `value` column on merge. We don't actually error but instead return `na_sentinel` after performing a merge against 0's. This was previous behavior.

But upon going through different possible categorical input combinations I realized that we can actually short-circuit the whole merging & sorting if there is a `ValueError` in type-cast, we just return a column filled with `na_sentinel` values.